modules/openapi-generator/src/main/resources/Java/libraries/aws-java-client/ApiClient.mustache

```diff
@@ -0,0 +1,593 @@
+    * @param queryParam The query parameters
+    * @param collectionQueryParams The collection query parameters
+    * @return The HTTP call
+    * @throws ApiException If fail to serialize the request body object
+    */
+    public Response buildCall(String path, String method, List<Pair> queryParam, List<Pair> collectionQueryParams) throws ApiException {
+        Request apiRequest = buildRequest(path, method, queryParam, collectionQueryParams);
+        return newCall(apiRequest);
+    }
+
+    /**
+    * Makes call to API and returns the Response
+    *
+    * @param apiRequest The request to the API
+    * @return The Response from the API
+    */
+    public Response newCall(Request apiRequest) {
+        Response response = new AmazonHttpClient(new ClientConfiguration())
+            .requestExecutionBuilder()
+            .executionContext(new ExecutionContext(true))
+            .request(apiRequest)
+            .errorResponseHandler(new SimpleAwsErrorHandler())
+            .execute(new HttpResponseHandler<String>() {
+                @Override
+                public String handle(HttpResponse response) throws Exception {
+                    String fromHttp = "";
+                    Scanner requestStream = new Scanner(response.getContent());
+                    while (requestStream.hasNext()) {
+                        fromHttp += requestStream.nextLine() + "\n";
+                    }
+                    return fromHttp;
+                }
+                @Override
+                public boolean needsConnectionLeftOpen() {
+                    return false;
+                }
+            });
+        return response;
+    }
+
+    /**
+    * Build an HTTP request with the given options.
+    *
+    * @param path The sub-path of the HTTP URL
+    * @param method The request method, one of "GET", "HEAD", "OPTIONS", "POST", "PUT", "PATCH" and "DELETE"
+    * @param queryParams The query parameters
+    * @param collectionQueryParams The collection query parameters
+    * @return The HTTP request
+    * @throws ApiException If fail to serialize the request body object
+    */
+    public Request buildRequest(String path, String method, List<Pair> queryParams, List<Pair> collectionQueryParams) throws ApiException {
+        final String url = buildUrl(path, queryParams, collectionQueryParams);
+
+        // Creating Request
+        Request apiRequest = new DefaultRequest<>("");
+        apiRequest.setHttpMethod(HttpMethodName.valueOf(method));
+
+        apiRequest.setEndpoint(URI.create(url));
+        apiRequest = signRequest(apiRequest);
+
+        return apiRequest;
+    }
+
+    /**
+    * Build an HTTP request with the given options.
+    *
+    * @param apiRequest The Request to the API which needs to be signed
+    * @return A signed Request
+    */
+    private Request signRequest(Request apiRequest) {
+        AWS4Signer signer = new AWS4Signer();
+        if (this.region != "") {
```

Comment:
region is string right?
You should check it with "".equals(this.region) or use StringUtils.isNotBlank(this.region) here